nixos/modules/programs/benchexec.nix

```diff
@@ -0,0 +1,59 @@
+{ lib
+, pkgs
+, config
+, options
+, ...
+}:
+let
+  cfg = config.programs.benchexec;
+  opt = options.programs.benchexec;
+in
+{
+  options.programs.benchexec = {
+    enable = lib.options.mkEnableOption "BenchExec";
+    package = lib.options.mkPackageOption pkgs "benchexec" { };
+
+    users = lib.options.mkOption {
+      type = lib.types.listOf lib.types.str;
+      description = lib.mdDoc ''
+        Users that intend to use BenchExec. Control group delegation will be configured via systemd.
+        For more information, see <https://github.com/sosy-lab/benchexec/blob/3.18/doc/INSTALL.md#setting-up-cgroups>.
+      '';
+      default = [ ];
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions = map
+      (user: {
+        assertion = config.users.users ? ${user};
+        message = ''
+          The user '${user}' intends to use BenchExec (via `${opt.users}`), but is not configured via `${options.users.users}`.
+        '';
+      })
+      cfg.users;
```

Comment:
Not supported, since don't know how (it's not possible?) to configure `Delegate=yes` for their units via configuration. Maybe this could be worked around by some magic that runs at system boot, but I do not want to go that far.

Reply:
Mutable users are now supported by adding their UID (as an integer) to `programs.benchexec.users`.